symengine/tests/basic/test_polynomial.cpp

```diff
@@ -335,6 +468,28 @@ TEST_CASE("UnivariatePolynomial get_args", "[UnivariatePolynomial]")
                               {one, mul(integer(2), x), pow(x, integer(2))}));
     REQUIRE(not vec_basic_eq_perm(
         a->get_args(), {one, mul(integer(3), x), pow(x, integer(2))}));
+
+    a = univariate_polynomial(x, {{0, 1}, {1, 1}, {2, 2}});
+    REQUIRE(vec_basic_eq_perm(a->get_args(),
+                              {one, x, mul(integer(2), pow(x, integer(2)))}));
+
+    a = univariate_polynomial(x, std::map<int, Expression>{});
+    REQUIRE(vec_basic_eq_perm(a->get_args(), {zero}));
+}
+
+TEST_CASE("UnivariatePolynomial max_coef", "[UnivariatePolynomial]")
+{
+    RCP<const Symbol> x = symbol("x");
+    RCP<const UnivariatePolynomial> a
+        = univariate_polynomial(x, {{0, 1}, {1, 2}, {2, 4}});
+    RCP<const UnivariatePolynomial> b
+        = univariate_polynomial(x, {{0, 2}, {1, 2}, {2, symbol("b")}});
+
+    REQUIRE(a->max_coef() == 4);
+    REQUIRE(not(a->max_coef() == 2));
+    REQUIRE(b->max_coef() == symbol("b"));
```

Comment:
Use `eq` here.

Error is due to `Catch` uses macros to take the LHS and RHS separately and will try to print each side with `<<` to show that RHS and LHS are different if there are different.